src/properties.c

```diff
@@ -0,0 +1,683 @@
+	 * pointer argument, since setter method usually makes its own copy of
+	 * the data. 
+	 *
+	 * We never set GtkSearchEntry::text - it is set when the user edits the
+	 * search entry.
+	 */
+	patt = gtk_editable_get_text( GTK_EDITABLE( search_entry ) );
+
+	/* Reuse the same buffer for the fuzzy matching algorithm.
+	 */
+	v = g_malloc( (strlen( patt ) + 1) * sizeof( guint ) );
+
+	/* Get a GList of GList of exact Match objects. There is an inner GList
+	 * for each property name in @field_list containing one or more exact
+	 * matching substrings. Each inner GList contains an exact Match for
+	 * each exactly matching substring within the same property name. 
+	 */
+	l = match_list( TRUE, p->field_list, patt, p->ignore_case, v );
+
+	/* If there were exact matches, add them to the UI. If there were no
+	 * exact matches, get inexact matches, sort them by
+	 * LEVENSHTEIN_DISTANCE, and add them to the UI.
+	 */
+	if( l )
+		/* Don't show the search warning, since there are exact matches
+		 * to show.
+		 */
+		gtk_widget_set_visible( p->search_warning, FALSE );
+	else {
+		/* Tell the user there are no exact matches.
+		 */
+		gtk_widget_set_visible( p->search_warning, TRUE );
+
+		/* Get a GList of GList of single inexact Match objects. There
+		 * is an inner GList for each property name in @field_list. Each
+		 * inner GList contains a single inexact match.
+		 */
+		l = match_list( FALSE, p->field_list, patt, p->ignore_case, v );
+
+		/* Sort by increasing Levenshtein Distance.
+		 */
+		l = g_list_sort( l, match_list_comp );
+
+		/* Truncate the list at NUM_INEXACT_MATCHES elements. Then,
+		 * fully clean up the truncated tail of @l, a GList of
+		 * GList of Match.
+		 */
+		t = g_list_nth( l, NUM_INEXACT_MATCHES );
+		if( t && t->prev ) {
+			t->prev->next = NULL;
+			match_list_free( t );
+		}
+	}
+
+	/* Add the matches to the UI.
+	 */
+	g_list_foreach( l, properties_append_field, p );
+
+	/* Clean up the buffer used by the fuzzy matching algorithm.
+	 */
+	g_free( v );
+
+	/* Fully clean up @l, a GList of GList of Match.
+	 */
+	match_list_free( l );
+
+	/* Clean up the array we got from vips_image_get_fields using the
+	 * method recommended by the VIPS docs (see vips_image_get_fields).
+	 */
+	g_strfreev( fields );
+}
+
+/* This is like the "destructor" method of GObject. It is called to clean up
+ * your object near the end of its lifetime.
+ *
+ * @p_	GObject * (Properties *)	this
+ *
+ * (GObject boilerplate).
+ */
+static void
+properties_dispose( GObject *p_ )
+{
+	GtkWidget *t;
+	Properties *p;
+
+#ifdef DEBUG
+	puts( "properties_dispose" );
+#endif
+
+	/* Cast to Properties *.
+	 */
+	p = VIPSDISP_PROPERTIES( p_ );
+
+	/* Clean up any existing children of @left_column and @right_column and
+	 * any existing elements of the GList @field_list.
+	 */
+	properties_clear_main_box( p );
+
+	/* Clean up the straggling child widget if there is one.
+	 */
+	if( (t = gtk_widget_get_first_child( GTK_WIDGET( p_ ) )) )
```

Comment:
You probably don't need to do this (I think?).

Reply:
Not having that if statement/block there creates a warning when the app closes about one of my widgets still having children to clean up. In general, I think I'm actually doing the minimal amount of cleanup code in terms of the UI, but I can check again.